### ISSUE TITLE
In 1.57.0 Style/RedundantDoubleSplatHashBraces autocorrect doesn't seem right

### ISSUE THREAD

**Issue by mikdiet** [rubocop/rubocop#12263]
I updated to 1.57.0, and in my code I see new Style/RedundantDoubleSplatHashBraces offence, on the line:

```
logger.info('ClassToLog', event: 'perform', **attributes.merge({ worker_id: worker.id }))
```
--------

## Expected behavior

I expect something like 

```
logger.info('ClassToLog', event: 'perform', worker_id: worker.id, **attributes)
```

## Actual behavior

I got

```
logger.info('ClassToLog', event: 'perform', attributes, { worker_id: worker.id })
```

And it doesn't even look like correct ruby expression.

## Steps to reproduce the problem

just run `rubocop -a` on above code

## RuboCop version

```
 » bundle exec rubocop -V
1.57.0 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-darwin21]
  - rubocop-factory_bot 2.24.0
  - rubocop-graphql 1.4.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.21.2
  - rubocop-rspec 2.24.1
  - rubocop-sorbet 0.7.4
```

**Comment by mikdiet** [comment]
In fact, the correct expectation should be

```
logger.info('ClassToLog', event: 'perform', **attributes, worker_id: worker.id)
```

I think, as if we have `worker_id` key in attributes, it should be overridden.

**Comment by mikdiet** [comment]
I also get ruby warning, when instead merging two hashes with the same key, I apply autocorrect for this cop, so it's probably should be marked as unsafe.

```
irb(main):004* def foo(**attrs)
irb(main):005*   puts attrs
irb(main):006> end
=> :foo
irb(main):007> foo(a: 1, **{ b: 2, c: 3}.merge({ c: 33 })) 
{:a=>1, :b=>2, :c=>33}
=> nil
irb(main):008> foo(a: 1, **{ b: 2, c: 3}, c: 33) 
(irb):8: warning: key :c is duplicated and overwritten on line 8
{:a=>1, :b=>2, :c=>33}
=> nil
```

**Comment by javierjulio** [comment]
This caused a regression for us if the double splat is used with an inline hash that happens to merge at the end so the double splat is still required. This is an example that triggers that warning when it should not:

```ruby
NotificationService.deliver(
  notification,
  user,
  **{
    a: 1,
    b: 2
  }.merge(notification_params)
)

def notification_params
  { c: 3, d: 4 }
end
```
